lib/AST/Decl.cpp

```diff
@@ -2221,6 +2221,15 @@ static Type mapSignatureFunctionType(ASTContext &ctx, Type type,
                                      bool isMethod,
                                      bool isInitializer,
                                      unsigned curryLevels) {
+  if (auto errorType = type->getAs<ErrorType>()) {
```

Comment:
Can you explain why this part was necessary?

Reply:
In `complete_where_clause` in the IDE tests, the `subscript` on line 106 results in an `ErrorType` (even though the underlying original type seems to be correct). When we tried to cast to `AnyFunctionType` on line 2244 below the compiler crashed. Should this be added as a comment and/or tracked as a bug?

Reply:
Should this method assume that `type->castTo<AnyFunctionType>()` will always succeed and leave any error checking to the caller instead?

Reply:
I see, I think is coming from `checkRedeclaration()`. Instead of destructuring the error type, how about just returning in this case.

Reply:
Sure, sounds good.